File: Find the Maximum Sum of Node Values.py
# https://leetcode.com/problems/find-the-maximum-sum-of-node-values/?envType=daily-question&envId=2024-05-19
# Find the Maximum Sum of Node Values

class Solution:
    def maximumValueSum(self, nums: List[int], k: int, edges: List[List[int]]) -> int:

        maxSum = sum(max(num, num ^ k) for num in nums)
        changedCount = sum((num ^ k) > num for num in nums)

        if changedCount % 2 == 0:
            return maxSum
        
        minChangeDiff = min(abs(num - (num ^ k)) for num in nums)

        return maxSum - minChangeDiff

File: Find the Maximum Sum of Node Values.js
// https://leetcode.com/problems/find-the-maximum-sum-of-node-values/?envType=daily-question&envId=2025-05-23
// Find the Maximum Sum of Node Values

/**
 * @param {number[]} nums
 * @param {number} k
 * @param {number[][]} edges
 * @return {number}
 */
var maximumValueSum = function(nums, k, edges) {

    const t = nums.map(i => (i ^ k) - i);

    t.sort((a, b) => b - a);

    let ans = nums.reduce((sum, num) => sum + num, 0);

    let index = 0;

    while (index + 1 < t.length) {
        const adder = t[index] + t[index + 1];
        if (adder > 0) {
            ans += adder;
        } else {
            break;
        }
        index += 2;
    };

    return ans;

};